compiler/rustc_parse/src/lexer/mod.rs

```diff
@@ -715,7 +719,12 @@ impl<'psess, 'src> StringReader<'psess, 'src> {
             } else {
                 None
             };
-            self.dcx().emit_err(errors::UnknownPrefix { span: prefix_span, prefix, sugg });
+            let err = errors::UnknownPrefix { span: prefix_span, prefix, sugg };
+            if silence {
+                self.dcx().create_err(err).delay_as_bug();
+            } else {
+                self.dcx().emit_err(err);
+            }
```

Comment:
I would probably structure it a bit differently, namely `let err = create_err(); if silence { err.delay_as_bug() } else { err.emit() }` but that's just a difference in taste. Definitely not blocking.